### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by delvedor**
As titled.

TODO:
- [x] Add explanation of what the `workbench` folder is and how to use it
- [x] Add a separate document that shows the entire process => failing test => fix input => run validation => green test

https://github.com/elastic/elastic-client-generator/pull/62 needs to be merged first.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,9 +159,210 @@ class IndexRequest<TDocument> extends RequestBase { ... }
...
+When you define a property the syntax is `propertyName: propertyType`.
+By default a property is required to exists. If you know that a property will not
+always be there, you can add a question mark juist before the column:
+
+```ts
+propertyRequired: string
+propertyOptional?: string
+```
+
+### A defintion is missing, how do I add it?
```
Comment (swallez) [inline README.md:321]: Typo

```suggestion
### A definition is missing, how do I add it?
```

File: README.md [hunk 2]
```diff
@@ -61,9 +159,210 @@ class IndexRequest<TDocument> extends RequestBase { ... }
...
+
+All the definitons are inside `specifications/specs` folder, you should explore its content and find the
```
Comment (swallez) [inline README.md:323]: Moar links (and  typos)

```suggestion
All the definitons are in the [`specifications/specs`](specifications/specs) folder, you should explore its content and find the
```

File: README.md [hunk 3]
```diff
@@ -61,9 +159,210 @@ class IndexRequest<TDocument> extends RequestBase { ... }
...
+
+### An endpoint definition is not correct, how do I fix it?
+
+All the endpoint definitons are inside `specifications/specs/json` folder, which contains a series of
+JSON files taken directly from the Elasticsearch rest-api-spec.
+You should copy from there the updated endpoint defintion and change it here.
+
+### The validation in broken on GitHub but works on my machine!
+
+Very likely thew recordings in our machine are stale, you can regenerate them
```
Comment (swallez) [inline README.md:346]: Typos

```suggestion
Very likely the recordings on your machine are stale, you can regenerate them
```

File: README.md [hunk 4]
```diff
@@ -61,9 +159,210 @@ class IndexRequest<TDocument> extends RequestBase { ... }
...
+
+```sh
+PULL_LATEST=true ./run-validations
+```
+
+### Which editor should I use?
+
+Any editor is fine, but to have a better development experience it should be configured
+to work with TypeScript. If you don't have it, [Visual Studio Code](https://code.visualstudio.com/)
+comes with TypeScript support out of the box.
```
Comment (swallez) [inline README.md:357]: Also mention IntelliJ that most Java developers (i.e. ES devs) use.

```suggestion
to work with TypeScript. [Visual Studio Code](https://code.visualstudio.com/) and
[IntelliJ IDEA](https://www.jetbrains.com/idea/) come with TypeScript support out of the box.
```

File: README.md [hunk 5]
```diff
@@ -1,19 +1,116 @@
...
+
+# the generated output can be found in ./output/schema/schema.json
+$ cat output/schema/schema.json
+```
+
+### Structure of the JSON representation
+
+The representation is a JSON object with two top level keys:
+
+```json
```
[the remark this answers is not in the document]
Comment (delvedor) [reply README.md:30]: Cool!

**Review by Mpdreamz — APPROVED** [review]
Maybe add a note on `realpath` that we have in clients-team that @dgieselaar also ran into on OSX?

Other than that LGTM.

**Pull request merged**